### ISSUE TITLE
space-before-function-parentheses errors on generator function expressions

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#1929]
```js
// test.js
function* foo () {}
var bar = function* () {};
```

```json
// config.json
{
	"ecmaFeatures": {
		"generators": true
	},
	"rules": {
		"generator-star": [2, "start"],
		"space-before-function-parentheses": 2
	}
}
```

```sh
$ eslint --reset --no-eslintrc --config config.json test.js

test.js
  3:18  error  Missing space before function parentheses  space-before-function-parentheses

✖ 1 problem (1 error, 0 warnings)
```

Moving this from #1920, as it's a separate issue. I propose that `space-before-function-parentheses` should not warn on generator function expressions at all, instead deferring to `generator-star`.

**Comment by nzakas** [comment]
Agreed
